.github/workflows/rust.yml

```diff
@@ -1,13 +1,20 @@
-name: Rust CI
+name: Rust CI/CD
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
Create another workflow file for CD, the triggers  are generally not the same. You run CI on every open pull request, and you run CD on merge. You would not want to run CD as soon as someone creates a PR.